packages/ti_eclecticiq/data_stream/outgoing_feed/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,420 @@
+---
+description: Pipeline for processing Device logs.
+processors:
+  # hard coded fields
+  - set:
+      field: ecs.version
+      value: 8.9.0
+  - set:
+      field: event.kind
+      value: enrichment
+  - set:
+      field: event.category
+      value: [threat]
+  - set:
+      field: event.type
+      value: [indicator]
+
+  # pre-processing
+  - rename:
+      field: message
+      target_field: event.original
+  - json:
+      field: event.original
+      tag: 'json_decoding'
+      target_field: json
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+
+  - drop:
+      description: Drop if extract type is not mapped.
+      if: "!['asn', 'domain', 'email', 'file', 'file-size', 'hash-md5', 'hash-md5', 'hash-sha1', 'hash-sha256', 'hash-sha384', 'hash-sha512', 'hash-ssdeep', 'ipv4', 'ipv4-cidr', 'ipv6', 'ipv6-cidr', 'mac-48', 'mutex', 'port', 'process', 'process-name', 'uri', 'winregistry', 'certificate-serial-number', 'malware', 'rule', 'user-agent', 'organization', 'email-subject', 'host', 'cve'].contains(ctx.json.type)"
+
+  # expand collapsed fields
+  - dot_expander:
+      path: json
+      field: source.names
+  - dot_expander:
+      path: json
+      field: meta.tags
+  - dot_expander:
+      path: json
+      field: meta.estimated_observed_time
+  - dot_expander:
+      path: json
+      field: meta.estimated_threat_start_time
+  - dot_expander:
+      path: json
+      field: meta.estimated_threat_end_time
+  - dot_expander:
+      path: json
+      field: meta.ingest_time
+
+  # base event fields
+  - set:
+      field: event.provider
+      value: '{{{json.source.names}}}'
+      if: "ctx.json.source?.names != null && ctx.json.source?.names != ''"
```

```suggestion
      if: "ctx.json.source?.names != null && ctx.json.source.names != ''"
```
Same for all other conditionals.